### PR TITLE
feat: knative service trait annotation support

### PR DESCRIPTION
**Pull request by realMartinez**
- Added route annotations implementation
    - Added unit tests
    - Added documentation

_CLI parameter usage example for clarity:_
`-t "knaive-service.annotations.'haproxy.router.openshift.io/balance'=true"`
_This can be repeated to add multiple annotations_

**Release Note**
```release-note
feat: added annotations support for knative service trait
```

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: pkg/trait/knative_service_test.go [hunk 1]
```diff
@@ -19,8 +19,10 @@ package trait
 
 import (
 	"path/filepath"
+	"reflect"
 	"testing"
 
+	routev1 "github.com/openshift/api/route/v1"
```
Comment (claudio4j) [inline pkg/trait/knative_service_test.go:25]: Is it correct to use the openshift route object to test the knative service annotation ?

File: pkg/trait/knative_service.go [hunk 2]
```diff
@@ -175,6 +178,8 @@ func (t *knativeServiceTrait) getServiceFor(e *Environment) (*serving.Service, e
 		for k, v := range e.Integration.Annotations {
 			serviceAnnotations[k] = v
```
Comment (claudio4j) [inline pkg/trait/knative_service.go:179]: It seems the current behavior to set the annotations in the integration and they are propagated to the knative service is actually right, isn't ?
If the user sets the knative service route annotation, then the integration annotations are not propagated to the knative service, which is not backward compatible.

File: pkg/apis/camel/v1/trait/knative_service.go [hunk 3]
```diff
@@ -26,6 +26,10 @@ package trait
 // +camel-k:trait=knative-service.
 type KnativeServiceTrait struct {
 	Trait `property:",squash" json:",inline"`
+	// The annotations added to route.
+	// This can be used to set knative service specific annotations
+	// CLI usage example: -t "route.annotations.'haproxy.router.openshift.io/balance'=true"
```
Comment (squakez) [inline pkg/apis/camel/v1/trait/knative_service.go:31]:
```suggestion
	// CLI usage example: -t "knative-service.annotations.'haproxy.router.openshift.io/balance'=true"
```

File: pkg/trait/knative_service.go (hunk 2 above)
Comment (squakez) [reply pkg/trait/knative_service.go:179]: Yes. We need to append the trait annotation after the Integration annotations are propagated.

File: pkg/trait/knative_service_test.go [hunk 4]
```diff
@@ -512,3 +514,26 @@ func createKnativeServiceTestEnvironment(t *testing.T, trait *traitv1.KnativeSer
 
 	return environment
 }
+func TestServiceAnnotation(t *testing.T) {
+	annotationsTest := map[string]string{"haproxy.router.openshift.io/balance": "true"}
+
+	environment := createTestRouteEnvironment(t, KnativeServiceTestName)
+	environment.Integration.Spec.Traits = v1.Traits{
+		Route: &traitv1.RouteTrait{
```
Comment (squakez) [inline pkg/trait/knative_service_test.go:522]: Why are you testing Route annotation? I think you need to test `KnativeServiceTrait` instead.

**Review by squakez — CHANGES_REQUESTED** [review]

**Comment by realMartinez** [comment]
I had a look at these issues, the annotations should be appending now and the test should use the correct environment as well. Doc is also displaying the example trait correctly.

File: pkg/trait/knative_service_test.go (hunk 1 above)
Comment (realMartinez) [reply pkg/trait/knative_service_test.go]: No I should have used serving instead

**Review by squakez — APPROVED** [review]

**Comment by squakez** [comment]
Also run a `make generate` in order to refresh the documentation/API accordingly.

**Pull request merged**